ambiguous reference in dog food script :dog:

The dog.csx script has an ambiguous reference:

ERROR: dog.csx(3,57): error CS0104: 'Verbosity' is an ambiguous reference between 'BauMSBuild.Verbosity' and 'BauNuGet.Verbosity'

Broken by: 48745e6dd6c4c389f91d9a3c2348b9c91284982c

--- a/baufile.csx
+++ b/baufile.csx
@@ -1,7 +1,7 @@
 // parameters
 var versionSuffix = Environment.GetEnvironmentVariable("VERSION_SUFFIX") ?? "-adhoc";
-var msBuildFileVerbosity = (Verbosity)Enum.Parse(typeof(Verbosity), Environment.GetEnvironmentVariable("MSBUILD_FILE_VERBOSITY") ?? "minimal", true);
-var nugetVerbosity = Environment.GetEnvironmentVariable("NUGET_VERBOSITY") ?? "quiet";
+var msBuildFileVerbosity = (BauMSBuild.Verbosity)Enum.Parse(typeof(BauMSBuild.Verbosity), Environment.GetEnvironmentVariable("MSBUILD_FILE_VERBOSITY") ?? "minimal", true);
+var nugetVerbosity = (BauNuGet.Verbosity)Enum.Parse(typeof(BauNuGet.Verbosity), Environment.GetEnvironmentVariable("NUGET_VERBOSITY") ?? "quiet", true);
 
 // solution specific variables
 var version = File.ReadAllText("src/CommonAssemblyInfo.cs").Split(new[] { "AssemblyInformationalVersion(\"" }, 2, StringSplitOptions.None).ElementAt(1).Split(new[] { '"' }).First();
@@ -30,7 +30,7 @@ bau
     msb.Properties = new { Configuration = "Release" };
     msb.MaxCpuCount = -1;
     msb.NodeReuse = false;
-    msb.Verbosity = Verbosity.Minimal;
+    msb.Verbosity = BauMSBuild.Verbosity.Minimal;
     msb.NoLogo = true;
     msb.FileLoggers.Add(new FileLogger
     {
@@ -46,9 +46,7 @@ bau
 
 .Task("clobber").DependsOn("clean").Do(() => DeleteDirectory(output))
 
-.Exec("restore").Do(exec => exec
-    .Run(nugetCommand)
-    .With("restore", solution))
+.NuGet("restore").Do(nuget => nuget.Restore(solution))
 
 .MSBuild("build").DependsOn("clean", "restore", "logs").Do(msb =>
 {
@@ -58,7 +56,7 @@ bau
     msb.Properties = new { Configuration = "Release" };
     msb.MaxCpuCount = -1;
     msb.NodeReuse = false;
-    msb.Verbosity = Verbosity.Minimal;
+    msb.Verbosity = BauMSBuild.Verbosity.Minimal;
     msb.NoLogo = true;
     msb.FileLoggers.Add(new FileLogger
     {
@@ -79,15 +77,15 @@ bau
 
 .Task("output").Do(() => CreateDirectory(output))
 
-.Exec("pack").DependsOn("build", "clobber", "output").Do(exec => exec
-    .Run(nugetCommand)
-    .With(
-        "pack", pack + ".csproj",
-        "-OutputDirectory", output,
-        "-Properties", "Configuration=Release",
-        "-IncludeReferencedProjects",
-        "-Verbosity", nugetVerbosity,
-        "-Version", version + versionSuffix))
+.NuGet("pack").DependsOn("build", "clobber", "output").Do(nuget => nuget
+    .Pack(
+        pack + ".csproj",
+        r => r
+            .WithOutputDirectory(output)
+            .WithProperty("Configuration","Release")
+            .WithIncludeReferencedProjects()
+            .WithVerbosity(nugetVerbosity)
+            .WithVersion(version + versionSuffix)))
 
 .Run();
 

--- a/dog.bat
+++ b/dog.bat
@@ -2,8 +2,8 @@
 
 ECHO [Dog] WARNING: Ensure you have built using baufile.csx before building with this!
 
-ECHO [Dog] Copying Bau.MSBuild from bin\Release...
-xcopy src\Bau.MSBuild\bin\Release\Bau.NuGet.dll packages\Bau.NuGet.0.1.0-beta01\lib\net45\ /Y /Q
+ECHO [Dog] Copying Bau.NuGet from bin\Release...
+xcopy src\Bau.NuGet\bin\Release\Bau.NuGet.dll packages\Bau.NuGet.0.1.0-alpha001\lib\net45\ /Y /Q
 
 ECHO [Dog] Running dog.csx...
 scriptcs dog.csx -- %*

--- a/dog.csx
+++ b/dog.csx
@@ -1,7 +1,7 @@
 // parameters
 var versionSuffix = Environment.GetEnvironmentVariable("VERSION_SUFFIX") ?? "-adhoc";
-var msBuildFileVerbosity = (Verbosity)Enum.Parse(typeof(Verbosity), Environment.GetEnvironmentVariable("MSBUILD_FILE_VERBOSITY") ?? "minimal", true);
-var nugetVerbosity = Environment.GetEnvironmentVariable("NUGET_VERBOSITY") ?? "quiet";
+var msBuildFileVerbosity = (BauMSBuild.Verbosity)Enum.Parse(typeof(BauMSBuild.Verbosity), Environment.GetEnvironmentVariable("MSBUILD_FILE_VERBOSITY") ?? "minimal", true);
+var nugetVerbosity = (BauNuGet.Verbosity)Enum.Parse(typeof(BauNuGet.Verbosity), Environment.GetEnvironmentVariable("NUGET_VERBOSITY") ?? "quiet", true);
 
 // solution specific variables
 var version = File.ReadAllText("src/CommonAssemblyInfo.cs").Split(new[] { "AssemblyInformationalVersion(\"" }, 2, StringSplitOptions.None).ElementAt(1).Split(new[] { '"' }).First();
@@ -12,8 +12,7 @@ var output = "artifacts/output";
 var tests = "artifacts/tests";
 var logs = "artifacts/logs";
 var unit = "src/test/Bau.NuGet.Test.Unit/bin/Release/Bau.NuGet.Test.Unit.dll";
-var packFolder = "src/Bau.NuGet/";
-var pack = "src/Bau.NuGet/Bau.Nuget";
+var pack = "src/Bau.NuGet/Bau.NuGet";
 
 // solution agnostic tasks
 var bau = Require<Bau>();
@@ -31,7 +30,7 @@ bau
     msb.Properties = new { Configuration = "Release" };
     msb.MaxCpuCount = -1;
     msb.NodeReuse = false;
-    msb.Verbosity = Verbosity.Minimal;
+    msb.Verbosity = BauMSBuild.Verbosity.Minimal;
     msb.NoLogo = true;
     msb.FileLoggers.Add(new FileLogger
     {
@@ -47,7 +46,7 @@ bau
 
 .Task("clobber").DependsOn("clean").Do(() => DeleteDirectory(output))
 
-.NuGet("restore").Do(nuget => nuget.Restore(Directory.EnumerateFiles("./","*.sln")))
+.NuGet("restore").Do(nuget => nuget.Restore(solution))
 
 .MSBuild("build").DependsOn("clean", "restore", "logs").Do(msb =>
 {
@@ -57,7 +56,7 @@ bau
     msb.Properties = new { Configuration = "Release" };
     msb.MaxCpuCount = -1;
     msb.NodeReuse = false;
-    msb.Verbosity = Verbosity.Minimal;
+    msb.Verbosity = BauMSBuild.Verbosity.Minimal;
     msb.NoLogo = true;
     msb.FileLoggers.Add(new FileLogger
     {
@@ -79,14 +78,14 @@ bau
 .Task("output").Do(() => CreateDirectory(output))
 
 .NuGet("pack").DependsOn("build", "clobber", "output").Do(nuget => nuget
-	.Pack(
-		Directory.EnumerateFiles(packFolder, "*.csproj"),
-		r => r
-			.WithOutputDirectory(output)
-			.WithProperty("Configuration","Release")
-			.WithIncludeReferencedProjects()
-			.WithVerbosity(nugetVerbosity)
-			.WithVersion(version + versionSuffix)))
+    .Pack(
+        pack + ".csproj",
+        r => r
+            .WithOutputDirectory(output)
+            .WithProperty("Configuration","Release")
+            .WithIncludeReferencedProjects()
+            .WithVerbosity(nugetVerbosity)
+            .WithVersion(version + versionSuffix)))
 
 .Run();
 

--- a/dog.debug.bat
+++ b/dog.debug.bat
@@ -3,8 +3,7 @@
 ECHO [Dog] WARNING: Ensure you have built using Debug config (e.g in Visual Studio) before building with this!
 
 ECHO [Dog] Copying Bau.NuGet from bin\Debug...
-xcopy src\Bau.NuGet\bin\Debug\Bau.NuGet.dll packages\Bau.NuGet.0.1.0-adhoc\lib\net45\ /Y /Q
-xcopy artifacts\output\Bau.NuGet.0.1.0-adhoc.nupkg packages\Bau.NuGet.0.1.0-adhoc\ /Y /Q
+xcopy src\Bau.NuGet\bin\Debug\Bau.NuGet.dll packages\Bau.NuGet.0.1.0-alpha001\lib\net45\ /Y /Q
 
 ECHO [Dog] Running dog.csx...
 scriptcs dog.csx -- %*

--- a/packages.config
+++ b/packages.config
@@ -3,6 +3,7 @@
   <package id="Bau" version="0.1.0-beta06" targetFramework="net45" />
   <package id="Bau.Exec" version="0.1.0-beta06" targetFramework="net45" />
   <package id="Bau.MSBuild" version="0.1.0-beta01" targetFramework="net45" />
+  <package id="Bau.NuGet" version="0.1.0-alpha001" targetFramework="net45" />
   <package id="Bau.Xunit" version="0.1.0-beta06" targetFramework="net45" />
   <package id="NuGet.CommandLine" version="2.8.3" />
   <package id="ScriptCs.Contracts" version="0.10.2" targetFramework="net45" />

